src/components/totals/TotalDistance/index.js

```diff
@@ -47,17 +48,34 @@ module.exports = React.createClass({
       isLoading: true
     });
 
-    campaigns.find(this.props.campaignUid, this.onSuccess);
+    campaigns.find(this.props.campaignUid, this.combineActivityData);
   },
 
-  onSuccess: function(result) {
-    var fitnessResults = result.campaign.fitness_activity_overview.run;
+  combineActivityData: function(result) {
```

Comment:
Hmm, I'll go over this with you tomorrow. We can use _.reduce to simplify this a lot. :-)